packages/relay-compiler/core/RelaySchemaUtils.js

```diff
@@ -214,6 +214,14 @@ function assertTypeWithFields(type: GraphQLType): GraphQLObjectType | GraphQLInt
   return type;
 }
 
+// TODO: Remove this when flow types for `getNullableType` in graphql-js
```

Comment:
I believe the type check in graphql-js handles more cases than this (via intersection type) - so it's possible whatever flow error was uncovered was real